swarm/api/content_type.go

```diff
@@ -0,0 +1,57 @@
+package api
+
+import (
+	"fmt"
+	"mime"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// detect content type by file content
+// otherwise by file extension
+// returns "application/octet-stream" in worst case
+func DetectContentType(file string) string {
+	var contentType = "application/octet-stream" // default value
```

Comment:
extract to `const MIME_OCTET_STREAM = "application/octet-stream"`, then reuse across the code (DRY)